WeakAuras/GenericTrigger.lua

```diff
@@ -882,10 +883,108 @@ function HandleUnitEvent(frame, event, unit, ...)
   WeakAuras.StopProfileSystem("generictrigger " .. event .. " " .. unit);
 end
 
+-- Periodic updates
+
+local function PeriodicStopCallback(data)
+  if data.periodic_context then
+    data.periodic_context.cancelled = true
+  end
+  data.periodic_context = nil
+end
+
+local function PeriodicStartCallback(id, triggernum, data)
+  -- tolerance is the maximum random delay for staggering updates
+  -- default will be +/- 5% of interval but no more than 50ms
+  local meanInterval = data.periodic_event.interval
+  local tolerance = data.periodic_event.tolerance or min(0.05, 0.05 * data.periodic_event.interval)
+
+  local periodic_context = {
+    id = id,
+    triggernum = triggernum,
+    delayMin = meanInterval - tolerance,
+    delayRange = 2 * tolerance,
+    cancelled = false,
+  }
+
+  local WeakAuras, C_TimerAfter, random = WeakAuras, C_Timer.After, random
```

Comment:
I wanted the random noise to occur at every tick (instead of just on load), which prevents the use of AceTimer or NewTicker. Adding noise on every tick should ensure more consistent update staggering.

Also, I looked at AceTimer before writing this and I think mine should be slightly more efficient since its API is less generic (fewer checks, no unpack call). Efficiency will be important if people do stuff like `PERIODIC_UPDATE:0.01`. I ran my last profiling test with `PERIODIC_UPDATE:0.01` and it gave 4.36% CPU, still far better than FRAME_UPDATE.